### DISCUSSION TITLE
umail library OSError: -202

### DISCUSSION THREAD

**Discussion by NahianMugdho** [micropython/micropython#10356] [Libraries & Drivers]
Hello, I'm using the umail library from the following source -
(https://github.com/shawwwn/uMail)

While I'm trying to run any of the examples that were given in the source, I'm getting the following error-

> Traceback (most recent call last):
> File "", line 2, in
> File "umail.py", line 30, in init
> OSError: -202

my code -
```
import umail
smtp = umail.SMTP('smtp.gmail.com', 587, username='my@gmail.com', password='mypassword')
smtp.to('someones@gmail.com')
smtp.send("This is an example.")
smtp.quit()

```

**Comment by th3w** [comment]
Did you setup and activate a network connection on the mcu  before you  run above code?

**Reply by NahianMugdho** [reply]
Hello, thanks for your reply. I have just written some code to set up wifi and added it to this example and now the OSerror is gone. But I'm getting a new error - 

> AssertionError: auth error 535'.['5.7.8 Username and Password not accepted. Learn more at', '5.7.8  https://support.google.com/mail/?p=BadCredentials 199-20020a6216d0000000b00580f224cdf6sm7014998pfw.203 - gsmtp']

**Comment by th3w** [comment]
What about using the option ssl=True and Port 465?

`smtp = umail.SMTP('smtp.gmail.com', 465, ssl=True, username='my@gmail.com', password='mypassword')`

**Reply by NahianMugdho** [reply]
The same error arrived. While searching on the internet, I found that I had to allow 'Less Secure Apps' on the sender's google ID. But from May 30, 2022, ​​Google no longer allows any less secure apps and there's no option to turn it on.

**Reply by davefes** [reply]
I use uMail on Gmail.  All that needs changing is:
```py
smtp = umail.SMTP('smtp.gmail.com', 587, username='mygmail.com', password='your 2 factor authentication')
```
